src/extension/serializer.ts

```diff
@@ -834,12 +823,21 @@ export class GrpcSerializer extends SerializerBase {
     metadata: { ['runme.dev/frontmatter']: string },
     kernel?: Kernel,
   ): Frontmatter {
-    const yamlDocs = YAML.parseAllDocuments(metadata['runme.dev/frontmatter'])
-    const data = (yamlDocs[0].toJS?.() || {}) as {
+    const rawFrontmatter = metadata['runme.dev/frontmatter']
+    let data: {
       runme: {
         id?: string
         version?: string
       }
+    } = { runme: {} }
+
+    if (rawFrontmatter) {
+      try {
+        const yamlDocs = YAML.parseAllDocuments(metadata['runme.dev/frontmatter'])
+        data = (yamlDocs[0].toJS?.() || {}) as typeof data
+      } catch (e) {
+        log.warn('failed to parse frontmatter')
```

Comment:
What about the error itself? Why not included it in the logging?

Reply:
Good point. I missed including it.